src/formats/png/filtering.zig

```diff
@@ -39,20 +40,23 @@ pub fn filter(writer: anytype, pixels: color.PixelStorage, filter_choice: Filter
 
     var y: usize = 0;
     while (y < header.height) : (y += 1) {
-        scanline = pixels.asBytes()[(y * scanline_len)..((y + 1) * scanline_len)];
```

Comment:
I started implementing this suggestion, and I'm not sure that this will save much complexity.

- We would still need a way to get a single scanline of data and then swap it's endianess
- `filter` would need to take an additional `allocator` parameter
- We would have to additionally allocate two scanlines of data and clean them up

This suggestion would simplify the loop let us remove `color.PixelStorage` from `filterChoiceHeuristic`, but I'm not sure if it's worth.

Reply:
If you are not sure I am OK that you leave it this way. You added a test to cover this case, and as long as we have good test coverage, we can try to simplify code in the future.